.changeset/lazy-nails-relax.md

```diff
@@ -0,0 +1,6 @@
+---
+'@shopify/polaris': minor
```

Comment:
I'm a bit late on this comment, but looking at these changes seems like this should be a `major` change since the current implementation is replacing the old component prop API.

If we want to keep it a minor, we should support both props, marking the old one as `@deprecated` and then do a followup PR to remove it in the next major.

@chloerice Should we revert for now so we aren't blocking the next Version Packages release?